lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -152,28 +153,59 @@ def add_order(post, options)
         order[:referenceCode] = options[:order_id] || generate_unique_id
         order[:description] = options[:description] || 'unspecified'
         order[:language] = 'en'
+        order[:shippingAddress] = shipping_address_fields(options) if options[:shipping_address]
         post[:transaction][:order] = order
       end
 
+      def add_payer(post, payment_method, options)
+        address = options[:billing_address]
+        payer = {}
+        payer[:fullName] = payment_method.name.strip
+        payer[:contactPhone] = address[:phone] if (address && address[:phone])
+        payer[:dniNumber] = options[:dni_number] if options[:dni_number]
+        payer[:dniType] = options[:dni_type] if options[:dni_type]
+        payer[:emailAddress] = options[:email] if options[:email]
+        payer[:birthdate] = options[:birth_date] if options[:birth_date] && options[:payment_country] == 'MX'
```

Comment:
Wondering if we should also check the billing address country for MX here?

Reply:
After inquiring with PayU, they indicated that the per-country required fields refers to the country of the merchant account, not the billing address. I agree it is strange.

Reply:
weird.